base/linalg/bidiag.jl

```diff
@@ -12,7 +12,7 @@ type Bidiagonal{T} <: AbstractMatrix{T}
         new(dv, ev, isupper)
     end
 end
-Bidiagonal{T}(dv::AbstractVector{T}, ev::AbstractVector{T}, isupper::Bool) = Bidiagonal{T}(dv, ev, isupper)
+Bidiagonal{T}(dv::AbstractVector{T}, ev::AbstractVector{T}, isupper::Bool) = Bidiagonal{T}(collect(dv), collect(ev), isupper)
```

Comment:
My reading of the `{T}` (which precedes me) is that it is meant to capture cases when `dv`, `ev` are of same type, to avoid dropping to the method below, which would incur the overhead of a `convert` that doesn't need to do anything. If the `{T}` is dropped, I wonder if this method would never be called and everything would drop down?